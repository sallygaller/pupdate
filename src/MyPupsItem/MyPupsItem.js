import React from "react";
import { Link } from "react-router-dom";

class MyPupsItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      checkDelete: false,
    };
  }

  render() {
    const pup = this.props.pup;
    return (
      <div>
        <Link to={`/pups/${pup.id}`}>
          <h3>{pup.name}</h3>
          <img
            className="MyPups-img"
            alt={`Profile of ${pup.name}`}
            src={`https://pupdate.s3-us-west-1.amazonaws.com/${pup.id}`}
          />
        </Link>
        <br></br>
        <Link to={`/edit/pups/${pup.id}`}>
          <button className="MyPups-profile">Edit Profile</button>
        </Link>
        {this.state.checkDelete ? (
          <div>
            Are you sure?
            <button
              className="MyPups-delete-profile"
              type="button"
              onClick={() => this.props.handleDeleteRequest(pup.id)}
            >
              Yes
            </button>
            <button
              className="MyPups-add"
              type="button"
              onClick={() => this.setState({ checkDelete: false })}
            >
              No
            </button>
          </div>
        ) : (
          <button
            className="MyPups-delete-profile"
            type="button"
            onClick={() => this.setState({ checkDelete: true })}
          >
            Delete Profile
          </button>
        )}
      </div>
    );
  }
}

export default MyPupsItem;
