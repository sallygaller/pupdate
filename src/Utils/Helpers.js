export default function playstyles(playstyle) {
  if (playstyle === "nervous") {
    return "I'm nervous or shy around other dogs.";
  }
  if (playstyle === "gentle") {
    return "I play gently.";
  }
  if (playstyle === "food-obsessed") {
    return "I'm food-obsessed!";
  }
  if (playstyle === "walks") {
    return "I like going on walks with my pup pals.";
  }
  if (playstyle === "parks") {
    return "I like going to dog parks with my pup pals.";
  }
  if (playstyle === "wrestling") {
    return "I like playfighting and wrestling.";
  }
  if (playstyle === "rambunctious") {
    return "I'm rambunctious and playful.";
  }
  if (playstyle === "ball-obsessed") {
    return "I love to play fetch!";
  }
}
