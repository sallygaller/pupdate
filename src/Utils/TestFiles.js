export const pupdates = [
  {
    id: 1,
    date: "2020-03-02T08:00:00.000Z",
    starttime: "09:30:00",
    endtime: "10:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park!",
    organizer: 1,
  },
  {
    id: 2,
    date: "2020-03-01T08:00:00.000Z",
    starttime: "15:30:00",
    endtime: "17:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park! I'll bring a frisbee",
    organizer: 1,
  },
];

export const pups = [
  {
    id: 1,
    name: "Eddie",
    breed: "dachshund",
    mix: true,
    age: "Adult",
    size: "M",
    nervous: true,
    rambunctious: false,
    gentle: true,
    wrestling: false,
    walks: true,
    parks: false,
    foodobsessed: true,
    ballobsessed: false,
    description:
      "Eddie can be a bit shy around other dogs at first, but he is really sweet! He loves going on long walks where there is a lot to sniff",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 1,
  },
  {
    id: 2,
    name: "Buddy",
    breed: "beagle",
    mix: true,
    age: "Adult",
    size: "L",
    nervous: false,
    rambunctious: true,
    gentle: false,
    wrestling: true,
    walks: false,
    parks: true,
    foodobsessed: false,
    ballobsessed: false,
    description:
      "Buddy can be a bit full on for some dogs, but he really just loves to play! He especially loves puppies",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 2,
  },
];
