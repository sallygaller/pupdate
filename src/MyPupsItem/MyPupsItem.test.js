import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import MyPupsItem from "./MyPupsItem";
import { pups } from "../Utils/TestFiles";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <MyPupsItem pup={pups[0]} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MyPupsItem pup={pups[0]} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
