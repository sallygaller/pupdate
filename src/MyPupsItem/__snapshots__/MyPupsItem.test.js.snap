// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <a
    href="/pups/1"
    onClick={[Function]}
  >
    <h3>
      Eddie
    </h3>
    <img
      alt="Profile of Eddie"
      className="MyPups-img"
      src="https://pupdate.s3-us-west-1.amazonaws.com/1"
    />
  </a>
  <br />
  <a
    href="/edit/pups/1"
    onClick={[Function]}
  >
    <button
      className="MyPups-profile"
    >
      Edit Profile
    </button>
  </a>
  <button
    className="MyPups-delete-profile"
    onClick={[Function]}
    type="button"
  >
    Delete Profile
  </button>
</div>
`;
